src/core/geometry/qgsgeometry.h

```diff
@@ -30,6 +30,7 @@ email                : morb at ozemail dot com dot au
 
 #include "qgis_core.h"
 #include "qgis_sip.h"
+#include "qgscoordinatereferencesystem.h"
```

Comment:
is this OK to add this include? it seems QgsGeometry was pretty light in terms of includes?

Reply:
I'd prefer not, but for different reasons 😁. There's a big warning at the top of this class that everything is Cartesian only, and I'm concerned if we start mixing in non Cartesian methods then the situation will get very murky. I'd lean toward putting this new function in a different utility class dedicated for ellipsoid aware geometry utils.

Reply:
I've dropped the ellipsoïd part.